Automattic/newspack-ios
# Housekeeping/update folder sort

This is a quick housekeeping PR to switch StoryFolders over to the new SortOrganizer, update tests, and remove the old SortRulesBook.

The implementation in the view controller is still mostly a placeholder just to illustrate the sorting behavior and will likely be improved on as the final UI is implemented. 

To Test:
- Make sure unit tests pass
- View the list of StoryFolders and make sure that sorting by date, name, ascending and descending still works, and that the sort choice is remembered.

@jleandroperez game for a quick review?
